config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OidcBackChannelLogoutTokenValidator.java

```diff
@@ -77,6 +78,9 @@ else if (events.get(BACK_CHANNEL_LOGOUT_EVENT) == null) {
 		if (issuer == null) {
 			errors.add(invalidLogoutToken("iss claim must not be null"));
 		}
+		else if (!StringUtils.hasText(this.issuer)) {
```

Comment:
I can see the value of doing it here. 

However, given that we already know this is invalid in the constructor, I'd prefer the check be there instead, preventing the object from getting constructed in an invalid state.

Perhaps something like this:

```java
String issuer = clientRegistration.getProviderDetails.getIsserUri();
Assert.hasText(issuer, "provider issuer cannot be null");
this.issuer = issuer;
...
```

Reply:
Also, that would more clearly indicate to the application that it should throw a 5xx instead of a 4xx, indicating that it is likely a server configuration problem, not a user request problem.